### ISSUE TITLE
Question about interpretation of rewards/environment evaluation

### ISSUE THREAD

**Issue by jkterry1** [LucasAlegre/sumo-rl#60]
So the reason me (and Ariel) have been working on the SUMO environment so heavily is that I'm trying to benchmark something sort of like a fancy new learning method on a bunch of different multi-agent environments, and SUMO (or traffic management in general) is a really desirable benchmark for us. I did a hyperparameter sweep of the learning method on my cluster last night on the ingolstadt7 map, and these are the ordered best rewards of all the training runs for different hyperparameters:

https://pastebin.com/Fh2hAQgK

The rewards appear to be railing at 0. I understand correctly from the description of rewards in the readme means that we learned an optimal policy, is that correct? Additionally, I was trying to see how our performance on it compared to results on the resco paper, however their results appear to be in different units than the reward we're using. What exactly are those units, and what's the easiest way to compute them?

**Comment by LucasAlegre** [comment]
> So the reason me (and Ariel) have been working on the SUMO environment so heavily is that I'm trying to benchmark something sort of like a fancy new learning method on a bunch of different multi-agent environments, and SUMO (or traffic management in general) is a really desirable benchmark for us. I did a hyperparameter sweep of the learning method on my cluster last night on the ingolstadt7 map, and these are the ordered best rewards of all the training runs for different hyperparameters:
> 
> https://pastebin.com/Fh2hAQgK
> 
> The rewards appear to be railing at 0. I understand correctly from the description of rewards in the readme means that we learned an optimal policy, is that correct? 

It is hard to know if it is really "optimal", but it seems that you learned a good policy!

> Additionally, I was trying to see how our performance on it compared to results on the resco paper, however their results appear to be in different units than the reward we're using. What exactly are those units, and what's the easiest way to compute them?

They are not plotting the reward, it seems they are plotting the "delay" and some other metrics in the tables. It seems they actually did not even fully define the reward function they used in the paper.

Notice that you can get the total number of stopped vehicles and the total waiting time for each time-step as a CSV file if you pass an 'out_csv_name' to the constructor. This is generally more informative than just plotting the total reward.

**Comment by jkterry1** [comment]
To confirm though, describing performance in these environments wouldn't be an unreasonable measure? All the readers in the traffic control community people aren't going to be expecting us to report delay instead or something?

**Comment by LucasAlegre** [comment]
Yes, in the traffic signal control community it is expected that you report traffic measures to evaluate your signal controller. More commonly used are Queue Lengths, Average Travel Times, Delay/Waiting Times, and so on. I did not even report total reward in my works on RL-based traffic signal control.

**Comment by LucasAlegre** [comment]
Unfortunately, things are a bit messy in the RL-based traffic signal control literature because it seems that everyone defines measures and experiments a bit differently. Total queue length (total number of stopped vehicles) is trivial and everyone computes the same way, but delay/waiting time can be defined differently. This is no problem if you define them clearly in your paper.

Ps: I have never used it, but SUMO has an argument to generate as output a file with average travel times, this could be useful.
